src/serverBuilder.ts

```diff
@@ -1,46 +1,50 @@
 import express from 'express';
 import bodyParser from 'body-parser';
-import { initAsync as validatorInit } from 'openapi-validator-middleware';
 import { container, inject, injectable } from 'tsyringe';
+import { middleware as OpenApiMiddleware } from 'express-openapi-validator';
 import { RequestLogger } from './common/middlewares/RequestLogger';
 import { ErrorHandler } from './common/middlewares/ErrorHandler';
 import { Services } from './common/constants';
-import { IConfig, ILogger } from './common/interfaces';
-import { routes } from './routes';
+import { IConfig } from './common/interfaces';
+import { entityRouterFactory } from './entity/routes/entityRouter';
+import { swaggerRouterFactory } from './common/routes/swagger';
 
 @injectable()
 export class ServerBuilder {
   private readonly serverInstance = express();
 
   public constructor(
-    @inject(Services.LOGGER) private readonly logger: ILogger,
     @inject(Services.CONFIG) private readonly config: IConfig,
     private readonly requestLogger: RequestLogger,
     private readonly errorHandler: ErrorHandler
   ) {
     this.serverInstance = express();
   }
 
-  public async build(): Promise<express.Application> {
-    //initiate swagger validator
-    await validatorInit(this.config.get('swaggerConfig.filePath'));
-
-    this.registerMiddleware();
+  public build(): express.Application {
+    this.registerPreRoutesMiddleware();
     this.buildRoutes();
+    this.registerPostRoutesMiddleware();
 
     return this.serverInstance;
   }
 
   private buildRoutes(): void {
-    routes.forEach((route) => {
-      const middlewares = route.middlewares ?? [];
-      this.serverInstance.use(route.path, ...middlewares, route.routerFactory(container));
-    });
+    this.serverInstance.use('/entity', entityRouterFactory(container));
+    this.serverInstance.use('/', swaggerRouterFactory(container));
   }
 
-  private registerMiddleware(): void {
-    this.serverInstance.use(bodyParser.urlencoded({ extended: true }));
+  private registerPreRoutesMiddleware(): void {
+    this.serverInstance.use(bodyParser.json());
+
+    const ignorePathRegex = new RegExp(`^${this.config.get<string>('swaggerConfig.basePath')}/.*`, 'i');
+    const apiSpecPath = this.config.get<string>('swaggerConfig.filePath');
+    this.serverInstance.use(OpenApiMiddleware({ apiSpec: apiSpecPath, validateRequests: true, ignorePaths: ignorePathRegex }));
```

Comment:
we should also ignore `liveness` and `readiness`, what do you think?

Reply:
Ignoring them is not necessary as they are not part of the express app, they are handled by the probe.